core/src/main/java/io/questdb/cairo/RebuildColumnBase.java

```diff
@@ -296,6 +300,11 @@ private void reindexOneOrAllColumns(
                 throw CairoException.nonCritical().put(unsupportedColumnMessage);
             }
         }
+        //To validate if the table has any index or not
```

Comment:
This check should be in the `if (columnIndex == REBUILD_ALL_COLUMNS)` block, under the loop.